### ISSUE TITLE
Required and Default don't work within Maps

### ISSUE THREAD

**Issue by pfavaloro** [dynamoosejs/dynamoose#674]
### Summary:

Required and default options don't work for fields within a map

#### Schema
```
new Schema({
  id: String,
  aMap: {
    type: 'Map',
    map: {
      a: Boolean,
      b: {type: String, required: true},
      c: {type: String, default: "a"},
    },
  },
} );
```

#### Model
```
{id: 'abcdefghijklm', aMap: {a: true}}
```

#### General
```
const streamSchema = new Schema({
  id: String,
  aMap: {
    type: 'Map',
    map: {
      a: Boolean,
      b: {type: String, required: true},
      c: {type: String, default: "a"},
    },
  },
} );
const Stream = dynamoose.model('test', streamSchema);



let a = new Stream({id: 'abcdefghijklm', aMap: {a: true}});
await a.save();
```

### Current output and behavior:

Saves:
 
```
{
  "aMap": {
    "a": true
  },
  "id": "abcdefghijklm"
}
```

### Expected output and behavior:

 Should error because `b` is missing, and if `b` were there, it should fill the default for `c`

### Environment:

Operating System: macOS High Sierra 
Operating System Version: 10.13.4
Node.js version (`node -v`):  v10.15.3
NPM version: (`npm -v`): 6.4.1
Dynamoose version: 1.11.0
Dynamoose Plugins: None

### Dynamoose Plugins:

- [ ] Yes, I believe that one or multiple 3rd party Dynamoose plugins are effecting this issue
- [X] No, I believe this issue is independent of any 3rd party Dynamoose plugins I'm using
- [ ] Unknown, I'm unsure if Dynamoose plugins are effecting this issue
- [X] I am not using any Dynamoose plugins

### Type (select 1):
- [X] Bug report
- [ ] Feature suggestion
- [ ] Question
- [ ] Other suggestion
- [ ] Something not listed here

### Other:
- [X] I have read through the Dynamoose documentation before posting this issue
- [X] I have searched through the GitHub issues (including closed issues) and pull requests to ensure this issue has not already been raised before
- [X] I have searched the internet and Stack Overflow to ensure this issue hasn't been raised or answered before
- [X] I have tested the code provided and am confident it doesn't work as intended
- [X] I have ensured that all of my plugins that I'm using with Dynamoose are listed above
- [X] I have filled out all fields above
- [X] I am running the latest version of Dynamoose

**Comment by kevinswarner** [comment]
I am also noticing that required and default do not work for lists of objects. For example, with the following schema...

```
const widgetSchema = new Schema({
  id: String,
  items: {
    type: 'list',
    list: [{
      id: {type: String},
      title: {type: String, default: 'Cool Widget'},
      description: {type: String, required: true}
    }],
  },
} );
const Widget = dynamoose.model('test', widgetSchema);

let widget = new Widget({id: 'foobar', items: [{id: 'item1'}]});
await widget.save();
```

1. Saving does not throw a "required" error for items[0].description.
2. Saving does not set default value for items[0].title

Perhaps this is not the expected behavior, but it would be helpful.

**Comment by rkrayovskyy** [comment]
+1 this would be helpful, have similar use case

**Comment by fishcharlie** [comment]
This should be fixed in version 2.0 beta 1. Let me know if you run into any more problems!
